cmd/zoekt-sourcegraph-indexserver/cleanup.go

```diff
@@ -71,19 +73,28 @@ func cleanup(indexDir string, repos []string, now time.Time) {
 
 		log.Printf("restoring shards from trash for %s", repo)
 		moveAll(indexDir, shards)
-		shardsLog(indexDir, "restore", shards)
+		shardsLog(indexDir, "restore", shards, repo)
 	}
 
-	// index: Move non-existant repos into trash
-	for _, shards := range index {
+	// index: Move non-existent repos into trash
+	for repo, shards := range index {
 		// Best-effort touch. If touch fails, we will just remove from the
 		// trash sooner.
 		for _, shard := range shards {
 			_ = os.Chtimes(shard.Path, now, now)
 		}
 
+		if tombstonesEnabled {
+			if len(shards) > 0 && strings.HasPrefix(filepath.Base(shards[0].Path), "compound-") {
+				shardsLog(indexDir, "set_tombstone", shards, repo)
+				if err := setTombstones(shards, repo); err != nil {
+					log.Printf("error setting tombstone for %s in %+v: %s\n", repo, shards, err)
```

Comment:
if we fail to set the tombstone, should we fallback to deleting the compound shard?

Reply:
alright, I also removed iterating over multiple shards. There should only ever be 1 when it comes to compound shards.